### PR TITLE
Include http:Headers parameter to websubhub:Service methods

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#1019](https://github.com/ballerina-platform/ballerina-standard-library/issues/1019)

### CODE REVIEW

File: websubhub-ballerina/tests/hub_with_header_details_test.bal [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    } else {
+        test:assertFail("Deregistration test failed");
+    }
+}
+
+function logHeaderDetails(http:Headers headers) returns error? {
+    log:print("Printing http header details");
+    foreach var headerName in headers.getHeaderNames() {
+        string headerVal = headerName + " : " + check headers.getHeader(headerName);
+        io:println(headerVal);
```
Comment (MadhukaHarith92) [inline websubhub-ballerina/tests/hub_with_header_details_test.bal:100]: Is this needed?

File: websubhub-ballerina/tests/hub_with_header_details_test.bal [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import ballerina/http;
+import ballerina/test;
+
+listener Listener hubWithHeaderDetailsListener = new(9095);
+
+var hubWithHeaderDetails = service object {
+
+    remote function onRegisterTopic(TopicRegistration message, http:Headers headers)
+                                returns TopicRegistrationSuccess {
+        log:print("Executing topic registration for ", req = message);
```
Comment (MadhukaHarith92) [inline websubhub-ballerina/tests/hub_with_header_details_test.bal:28]:
```suggestion
        log:print("Executing topic registration", message = message);
```

File: websubhub-ballerina/tests/hub_with_header_details_test.bal [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                body: <map<string>>{
+                       isSuccess: "true"
+                    }
+        };
+        return successResult;
+    }
+
+    remote function onDeregisterTopic(TopicDeregistration message, http:Headers headers)
+                        returns TopicDeregistrationSuccess {
+        log:print("Executing topic de-registration for ", req = message);
```
Comment (MadhukaHarith92) [inline websubhub-ballerina/tests/hub_with_header_details_test.bal:40]:
```suggestion
        log:print("Executing topic de-registration", message = message);
```

File: websubhub-ballerina/utils.bal [hunk 4]
```diff
@@ -71,8 +73,15 @@ function processSubscriptionRequestAndRespond(http:Request request, http:Caller
...
     var hubLeaseSeconds = params[HUB_LEASE_SECONDS];
-    if (hubLeaseSeconds is () || 'int:fromString(hubLeaseSeconds) == 0) {
+    if (hubLeaseSeconds is ()) {
         hubLeaseSeconds = defaultHubLeaseSeconds.toString();
+    } else {
+        var retrievedLeaseSeconds = 'int:fromString(hubLeaseSeconds);
+        if (retrievedLeaseSeconds is error) {
+            hubLeaseSeconds = defaultHubLeaseSeconds.toString();
+        } else if (retrievedLeaseSeconds == 0) {
+            hubLeaseSeconds = defaultHubLeaseSeconds.toString();
```
Comment (MadhukaHarith92) [inline websubhub-ballerina/utils.bal:83]: Both `if` and `else if` conditions do the same thing isn't it? If that's the case can we simplify it as follows?
```
if (retrievedLeaseSeconds is error || retrievedLeaseSeconds == 0)
```

**Review by MadhukaHarith92 — COMMENTED** [review]
Added few comments.

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**